src/pages/ServiceGraph/SummaryPanelEdge.tsx

```diff
@@ -33,82 +24,58 @@ export default class SummaryPanelEdge extends React.Component<SummaryPanelPropTy
-      destNamespace: destSplit[1],
-      destVersion: dest.data('version'),
       reqRates: [],
       errRates: []
     };
   }
 
   componentDidMount() {
-    const options = {
-      version: this.state.destVersion,
-      'byLabelsIn[]': 'source_service,source_version',
-      rateInterval: this.props.rateInterval
-    };
-    API.getServiceMetrics(this.state.destNamespace, this.state.destServiceName, options)
-      .then(response => {
-        const data: M.Metrics = response['data'];
-        const metrics: Map<String, M.MetricGroup> = data.metrics;
-        const reqRates = this.getRates(metrics['request_count_in'], 'RPS');
-        const errRates = this.getRates(metrics['request_error_count_in'], 'Error');
+    this.updateRpsChart(this.props);
+  }
 
-        this.setState({
-          loading: false,
-          reqRates: reqRates,
-          errRates: errRates
-        });
-      })
-      .catch(error => {
-        this.setState({ loading: false });
-        console.error(error);
-        // this.props.onError(error);
-      });
+  componentWillReceiveProps(nextProps: SummaryPanelPropType) {
+    if (nextProps.data.summaryTarget !== this.props.data.summaryTarget) {
+      this.updateRpsChart(nextProps);
+    }
   }
 
   render() {
     const edge = this.props.data.summaryTarget;
+    const source = edge.source();
+    const sourceService = source.data('service');
+    const sourceVersion = source.data('version');
+    const sourceSplit = sourceService.split('.');
+    const sourceServiceName = sourceSplit[0];
+    const sourceNamespace = sourceSplit.length < 2 ? 'unknown' : sourceSplit[1];
+    const dest = edge.target();
+    const destService = dest.data('service');
+    const destVersion = dest.data('version');
+    const destSplit = dest.data('service').split('.');
+    const destServiceName = destSplit[0];
+    const destNamespace = destSplit[1];
     const rate = this.safeRate(edge.data('rate'));
     const rate3xx = this.safeRate(edge.data('rate3XX'));
     const rate4xx = this.safeRate(edge.data('rate4XX'));
     const rate5xx = this.safeRate(edge.data('rate5XX'));
-    const sourceLink = (
-      <a href={`../namespaces/${this.state.sourceNamespace}/services/${this.state.sourceService}`}>
-        {this.state.sourceServiceName}
-      </a>
-    );
-    const destLink = (
-      <a href={`../namespaces/${this.state.destNamespace}/services/${this.state.destService}`}>
-        {this.state.destServiceName}
-      </a>
-    );
+    const sourceLink = <a href={`../namespaces/${sourceNamespace}/services/${sourceService}`}>{sourceServiceName}</a>;
+    const destLink = <a href={`../namespaces/${destNamespace}/services/${destService}`}>{destServiceName}</a>;
 
     return (
       <div className="panel panel-default" style={SummaryPanelEdge.panelStyle}>
-        <div className="panel-heading">Edge Source: {sourceLink}</div>
+        <div hidden={sourceServiceName === 'unknown'} className="panel-heading">
```

Comment:
Not a blocker, but a suggesetion: Perhaps this can take advantage of conditional rendering instead of injecting two blocks: https://reactjs.org/docs/conditional-rendering.html#inline-if-else-with-conditional-operator